gems/aws-sigv4/sig/signer.rbs

```diff
@@ -0,0 +1,50 @@
+module Aws
+  module Sigv4
+    class Signer
+      def initialize: (
+        service: String,
+        region: String,
+        ?access_key_id: String,
+        ?secret_access_key: String,
+        ?session_token: String,
+        ?credentials: untyped,
+        ?credentials_provider: untyped,
```

Comment:
I noticed that `aws-sigv4/credentials.rb` is marked `@api private`, so I removed its RBS definitions. Do you think it’s better to restore them or just define an interface (requiring `#credentials` returning access keys, etc.) directly in `signer.rbs`? I’m unsure which approach is more suitable, given the private status. Any thoughts?

Reply:
Yes, this is true. I think `sigv4/credentials.rb` serves as an intermediary. Technically these would be `Aws::Credentials` from aws-sdk-core. We can leave these untyped for now, and if you're interested in adding rbs for credentials in aws-sdk-core, we can fill these in. I'm not sure how rbs works with across packages, since aws-sigv4 does not depend on aws-sdk-core.

Reply:
So you’re suggesting that if we create RBS files corresponding to `gems/aws-sdk-core/lib/aws-sdk-core/credentials.rb` and `gems/aws-sdk-core/lib/aws-sdk-core/credential_provider.rb`, we could then make use of those type definitions, right? For now, I'll keep it `untyped`, but afterward I'm thinking of adding RBS definitions for `aws-sdk-core` credentials. How does that sound?

Reply:
While in most use cases, `Aws::Credentials` from `aws-sdk-core` would be used, this library is intended as a stand alone and doesn't depend on aws-sdk-core - I think from a typing perspective maybe we should define a credentials interface (ie: any object that responds to access_key_id, secret_access_key and session_token) since thats actually what is required.  And probably an interface for credentials provider as well.